scipy/interpolate/interpolate.py

```diff
@@ -340,10 +340,12 @@ class interp1d(_Interpolator1D):
     Parameters
     ----------
     x : (N,) array_like
-        A 1-D array of real values.
+        A 1-D array of real values. These values must be convertible to `float`
+        values.
```

Comment:
Not sure about this phrasing, sounds real and float the same to me. And it's not any more discoverable to the user who tried to interpolate the datetime.

Reply:
Thank you for reviewing.
> Not sure about this phrasing, sounds real and float the same to me.
OK. So, what do you think this phrasing? "A 1-D array of values which are convertible to `float`"

> And it's not any more discoverable to the user who tried to interpolate the datetime.
I'm sorry I am not sure your suggestion.. Should we explain datetime is not acceptable?

> I think it's better to add a sentence to the Notes section.
OK. I will try to add it in the next commit.

Reply:
I think the original phrasing was fine *if* accompanied by an explicit note in the Notes section (which will be indeed great to add).

Re datetime --- maybe there's a value in a worked example. It'll probably need to convert the datetime objects to a common dtype, subtract the initial value (all this to avoid overflows for large integers), interpolate and convert back to datetimes. Maybe it's too much for a docstring example, and the whole thing is better suited for a tutorial.

Reply:
OK. Thank you for your comments. I added Note section. PTAL.